node/src/types/transaction/deploy_or_transaction_hash.rs

```diff
@@ -0,0 +1,50 @@
+use std::hash::Hash;
+
+use datasize::DataSize;
+use derive_more::Display;
+
+use casper_types::{Transaction, TransactionHash, TransactionV1Hash};
+
+use super::{transaction_v1::TransactionV1OrTransferV1Hash, DeployOrTransferHash};
+
+#[derive(Copy, Clone, Display, Debug, Ord, PartialOrd, Eq, PartialEq, Hash, DataSize)]
+pub(crate) enum DeployOrTransactionHash {
+    #[display(fmt = "deploy {}", _0)]
+    Deploy(DeployOrTransferHash),
+    #[display(fmt = "transaction {}", _0)]
+    V1(TransactionV1OrTransferV1Hash),
+}
```

Comment:
I think maybe rather than having `DeployOrTransactionHash`, `DeployOrTransferHash` and `TransactionV1OrTransferV1Hash`, we could replace them all with something like
```rust
pub struct CategorizedTransactionHash {
    hash: TransactionHash,
    category: TransactionV1Category
}
```
Not sure though - maybe we have other uses for these three enums?

Reply:
The `TransactionV1OrTransferV1Hash` has been removed in 939e54ec4861eb7ef8153f04b3d1a165beab6ac7.

The outer enum is still there, since block validator needs the distinction between deploy and transfer for legacy transactions. If the assumption is that going forward block validator only deals with new transactions, we can clean this up further (cc: @Fraser999)